server/src/daemon/service.rs

```diff
@@ -24,11 +24,33 @@ use super::{
     collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
+impl ServiceManager {
+    pub fn new() -> Self {
+        Self {
+            sessions: HashMap::new(),
+        }
+    }
+
+    pub fn sessions(&self) -> &HashMap<OwnedObjectPath, Session> {
+        &self.sessions
+    }
+
+    pub fn set_sessions(&mut self, path: OwnedObjectPath, session: Session) {
+        self.sessions.insert(path, session);
+    }
+}
+
 #[derive(Debug)]
 pub struct Service {
     collections: Vec<Collection>,
     keyring: Arc<Keyring>,
     cnx: Mutex<Option<zbus::Connection>>,
+    manager: Arc<ServiceManager>,
```

Comment:
Is this approach looks clean to you? For `Service` it make sense. But for `Collection` and `Item`, I'm not so sure.

Reply:
> For Service it make sense. But for Collection and Item, I'm not so sure

You might need it there as well. You get the session path in get secrets so you have to match the path back to the already open session.

Although, not useful right now without making use of it.

Reply:
yeah, got it. Thanks!